### DISCUSSION TITLE
Laragon can serve HTML but not PHP

### DISCUSSION THREAD

**Discussion by mawg** [leokhoa/laragon#742] [General]
Any idea why?   `php -v` form the Laragon terminal works fine, but `http://localhost/phpinfo.php won't serve.

Btw, Apache seems to be always running and not deselectable, do I also need nginx? I haven't used it so far,and am not really sure what it's for. 

![image](https://github.com/leokhoa/laragon/assets/2999044/d2ea1ff7-dd5b-435e-9285-8a80e13301d7)

**Comment by Pen-y-Fan** [comment]
NGINX needs to be set on a different port e.g.

<img width="399" alt="image" src="https://github.com/leokhoa/laragon/assets/40126936/f2c6f76b-62f6-4edb-a938-6cc8da3d1e8c">

When you select NGINX you can deselect Apache. If you want NGINX on 80/443 change Apache to 8080/8443 first, then deselect. It's not possible to have both using 80/443 at the same time.

To enable SSL see my blog [How to set up Laragon on a new Windows computer (part 4) - Enable security](https://pen-y-fan.github.io/2022/11/10/how-to-set-up-laragon-on-a-new-windows-computer-part-4/)

To see phpinfo, go to localhost and click the info link, the URL is http://localhost/?q=info

**Reply by mawg** [reply]
Thanks *very* much. And that is a great series of blog posts (I will also be looking at the rest of your blog).

I am only interested in the portable version, as I seem to have to reinstall Windows every few months, and it's a real PITA; hence, I use as many portable apps as possible. XAMPP would do the trick, I guess, but I thought that I would give Laragon a whirl, for features such as ease of switching between PGP versions.

I put Apache and Nginx on different ports, but still can't serve PHP pages. Could the problem be with Apache? It seems suspicious that it is greyed out:

![image](https://github.com/leokhoa/laragon/assets/2999044/4d780913-dd2b-4625-b91b-de98c6a7db8b)

Any idea what's wrong there?

------------------------------------------------------------------------------------------

Wait!! Does Laragon portable even *include* Apache??

Laragon Full (64-bit): Apache 2.4, Nginx, MySQL 8, PHP 8, Redis, Memcached, Node.js 18, npm, git
[Download Laragon - Portable (38 MB)](https://github.com/leokhoa/laragon/archive/refs/tags/6.0.0.zip)

Laragon Portable: PHP 5.4, MySQL 5.1, bitmana - Good for getting started with PHP, then you can add newer versions of PHP/MySQL easily later using [“Tools > Quick add”](https://laragon.org/docs/quick-add.html)

So, maybe my question is "how to add Apache to Laragon portable"?

**Reply by Pen-y-Fan** [reply]
The full version of laragon is also portable, you may be better off starting with the full version and backing it up. To restore Laragon, on a new computer, you will also need to install Visual C++ Redistributable Visual Studio 2015-2022  (VS16 or VS17) runtime.

The portable version ships with NGINX, and Apache can be installed from [Apache lounge](https://www.apachelounge.com/download/), the links to Visual C++ Redistributable Visual Studio 2015-2022 are on this page too. 

MySQL community edition can also be downloaded and manually installed.

When unzipping the downloads take note of the structure of the existing apps and ensure the new ones are at the same folder level. e.g. watch for C:\laragon\bin\apache\httpd-2.4.54-win64-VS16\bin\httpd.exe, sometimes when unzipping an additional **apache** folder is created C:\laragon\bin\apache\apache\....

My recommendation is to install the full version and back it up.
